pkg/cloudprovider/provider/azure/types/types.go

```diff
@@ -36,8 +36,10 @@ type RawConfig struct {
 	AvailabilitySet   providerconfigtypes.ConfigVarString `json:"availabilitySet"`
 	SecurityGroupName providerconfigtypes.ConfigVarString `json:"securityGroupName"`
 
-	AssignPublicIP   providerconfigtypes.ConfigVarBool   `json:"assignPublicIP"`
-	Tags             map[string]string                   `json:"tags,omitempty"`
 	ImageID          providerconfigtypes.ConfigVarString `json:"imageID"`
 	RHSMOfflineToken providerconfigtypes.ConfigVarString `json:"rhsmOfflineToken,omitempty"`
+	OSDiskSize       providerconfigtypes.ConfigVarString `json:"osDiskSize"`
+	DataDiskSize     providerconfigtypes.ConfigVarString `json:"dataDiskSize"`
```

Comment:
This field should be int instead of having it as a string and then need to convert it to int again :)